### ISSUE TITLE
Add support for ember-decorators

### ISSUE THREAD

**Issue by rogatty** [Turbo87/intellij-emberjs#214]
If you want to use JS native classes you also need to use http://ember-decorators.github.io/ember-decorators/latest/docs. Currently, this plugin doesn't understand what the decorators mean. When using the `@service` decorator, there is no lightning bolt icon linking to the service, unlike when using `serviceName: service()`.

![screenshot](https://user-images.githubusercontent.com/7030884/42309406-c37dfeb8-8038-11e8-89a7-5b2fae851115.png)

**Comment by Turbo87** [comment]
@rogatty that would be nice! can you work on adding support for that?

**Comment by rogatty** [comment]
I have no experience with Kotlin or IntelliJ plugins but I'll try to poke it with a stick if I find the time for it.
I opened this issue mostly to let you know that there is a new Ember feature that will be more and more popular as people migrate to the native classes. If I don't get to it first, I hope the hundreds of :+1: :+1: :+1: on the original post will get your attention :wink:

**Comment by Turbo87** [comment]
> there is a new Ember feature

it's not an Ember feature yet though, only an addon that promotes an unstable ES proposal. once this gets into mainstream Ember I could have a look, but for now it's not high on my todo list 😉

**Comment by robclancy** [comment]
Octane is almost done and this addon is very outdated compared to it. This is probably the main feature that is needed though.

**Comment by Turbo87** [comment]
@robclancy I agree, would be great if someone could find the time to work on this.
